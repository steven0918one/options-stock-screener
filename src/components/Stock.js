import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function Stock({ stock }) {
  let source = '';
  if (
    [
      'NYSE Arca',
      'NYSE American',
      'NYSE',
      'New York Stock Exchange',
      'NYSEArca',
    ].includes(stock.exchange)
  ) {
    source = 'interesting-stocks';
  } else if (
    [
      'Nasdaq Global Select',
      'NASDAQ Global Market',
      'NASDAQ Capital Market',
      'Nasdaq',
      'NasdaqGM',
      'NasdaqCM',
      'NasdaqGS',
    ].includes(stock.exchange)
  ) {
    source = 'most-gainer';
  } else if (
    ['Paris', 'Amsterdam', 'Brussels', 'Lisbon', 'Toronto', 'Swiss'].includes(
      stock.exchange,
    )
  ) {
    source = 'most-loser';
  } else {
    source = 'most-active';
  }
  return (
    <div className={`info-box ${source}-box`} id={stock.symbol}>
      <Link to={`/details/${stock.symbol}`}>
        <p className="info-box-company" id={stock.symbol}>
          {stock.name}
        </p>
        <p className="info-box-ticker" id={stock.symbol}>
          {stock.symbol}
        </p>
        <div className="info-box-price-container" id={stock.symbol}>
          <span className="info-box-price" id={stock.symbol}>
            $
            {stock.price}
          </span>
        </div>
      </Link>
    </div>
  );
}

Stock.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  stock: PropTypes.object.isRequired,
};

export default Stock;
