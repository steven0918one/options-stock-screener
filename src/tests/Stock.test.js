import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Stock from '../components/Stock';
import Wrapper from './Wrapper';

Enzyme.configure({ adapter: new Adapter() });

test('Stock should match snapshot', () => {
  const stock = {
    exchange: 'Nasdaq',
    symbol: 'AAPL',
    name: 'Apple Inc.',
    price: '320',
  };
  const component = renderer.create(
    <Wrapper>
      <Stock stock={stock} />
    </Wrapper>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Stock component', () => {
  let wrapper;
  const stock = {
    exchange: 'Nasdaq',
    symbol: 'AAPL',
    name: 'Apple Inc.',
    price: '320',
  };
  beforeEach(() => {
    wrapper = mount(
      <Wrapper>
        <Stock stock={stock} />
      </Wrapper>,
    );
  });
  it('is receiving and running JS', () => {
    expect(2 + 3).toEqual(5);
  });

  it('renders', () => {
    expect(wrapper).not.toBeNull();
  });

  it('Displays stock name', () => {
    expect(wrapper.find('.info-box-company').text()).toEqual('Apple Inc.');
  });

  it('Displays stock symbol', () => {
    expect(wrapper.find('.info-box-ticker').text()).toEqual('AAPL');
  });

  it('Displays stock price', () => {
    expect(wrapper.find('.info-box-price').text()).toEqual('$320');
  });
});
