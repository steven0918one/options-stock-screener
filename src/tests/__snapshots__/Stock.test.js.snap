// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock is rendered 1`] = `
<div
  className="info-box most-gainer-box"
  id="AAPL"
>
  <a
    href="/details/AAPL"
    onClick={[Function]}
  >
    <p
      className="info-box-company"
      id="AAPL"
    >
      Apple Inc.
    </p>
    <p
      className="info-box-ticker"
      id="AAPL"
    >
      AAPL
    </p>
    <div
      className="info-box-price-container"
      id="AAPL"
    >
      <span
        className="info-box-price"
        id="AAPL"
      >
        $
        320
      </span>
    </div>
  </a>
</div>
`;

exports[`Stock should match snapshot 1`] = `
<div
  className="info-box most-gainer-box"
  id="AAPL"
>
  <a
    href="/details/AAPL"
    onClick={[Function]}
  >
    <p
      className="info-box-company"
      id="AAPL"
    >
      Apple Inc.
    </p>
    <p
      className="info-box-ticker"
      id="AAPL"
    >
      AAPL
    </p>
    <div
      className="info-box-price-container"
      id="AAPL"
    >
      <span
        className="info-box-price"
        id="AAPL"
      >
        $
        320
      </span>
    </div>
  </a>
</div>
`;
